backend/db/analysis.py

```diff
@@ -5,62 +5,77 @@
 from datetime import datetime
 from pathlib import Path
 from pprint import pprint
-from typing import Dict, List
+from typing import Any, Dict, List, Union
 import dateutil.parser as dp
 import pandas as pd
+from jinja2 import Template
 from pandas import Series
 
+
 THIS_DIR = os.path.dirname(__file__)
 PROJECT_ROOT = Path(THIS_DIR).parent.parent
 sys.path.insert(0, str(PROJECT_ROOT))
+from backend.db.config import DOCS_DIR
 from backend.db.initialize import SCHEMA
-from backend.db.utils import get_db_connection, insert_from_dict, list_tables, run_sql
+from backend.db.utils import get_db_connection, insert_from_dict, list_tables, run_sql, sql_in
 
 COUNTS_OVER_TIME_OPTIONS = [
-    'print_counts_table',
-    'print_delta_table',
+    'counts_table',
+    'delta_table',
     # 'save_delta_viz'
 ]
+DOCS_PATH = os.path.join(DOCS_DIR, 'backend', 'db', 'analysis.md')
+DOCS_JINJA = """# DB row counts
+## Counts over time
+{{ counts_markdown_table }}
+
+## Deltas over time
+{{ deltas_markdown_table }}"""
 
 
-# TODO: automatically compare to most recent backup. currently fixed to a specific schema
 def counts_compare_schemas(
```

Comment:
@Sigfried Now `counts_compare_schemas()` automatically compares to the most recent backup. Previously you'd have to pass the compare schema explicitly.

Reply:
Also this function now uses my accurate counts method, rather than the approximate method you wrote before (or the other one which was bugged for some reason; maybe just in SqlAlchemy. It might have been that `%` needed to change to `%%`).

Reply:
The output is also much cleaner now; it's a table rather than multiple rows of tuples. And I added a `delta` column.